tests/integration/cartography/intel/spacelift/test_spacelift.py

```diff
@@ -0,0 +1,281 @@
+TEST_ACCOUNT_ID = "test-account-123"
+
+
+@patch("cartography.intel.spacelift.runs.get_entities")
+@patch("cartography.intel.spacelift.runs.get_runs")
+@patch("cartography.intel.spacelift.workers.get_workers")
+@patch("cartography.intel.spacelift.workerpools.get_worker_pools")
+@patch("cartography.intel.spacelift.stacks.get_stacks")
+@patch("cartography.intel.spacelift.spaces.get_spaces")
+@patch("cartography.intel.spacelift.account.get_account")
+def test_spacelift_end_to_end(
+    mock_get_account,
+    mock_get_spaces,
+    mock_get_stacks,
+    mock_get_worker_pools,
+    mock_get_workers,
+    mock_get_runs,
+    mock_get_entities,
+    neo4j_session,
+):
+    """
+    End-to-end integration test for Spacelift module.
+    Tests syncing of all Spacelift resources and their relationships,
+    including Run-[:AFFECTS]->EC2Instance relationships.
+    """
+    # Mock all API calls using the mock data file
+    mock_get_account.return_value = ACCOUNT_DATA["data"]["account"]
+    mock_get_spaces.return_value = SPACES_DATA["data"]["spaces"]
+    mock_get_stacks.return_value = STACKS_DATA["data"]["stacks"]
+    mock_get_worker_pools.return_value = WORKER_POOLS_DATA["data"]["workerPools"]
+
+    # get_workers returns flattened workers with workerPool field added
+    mock_workers_flattened = []
+    for pool in WORKERS_DATA["data"]["workerPools"]:
+        for worker in pool.get("workers", []):
+            worker_copy = worker.copy()
+            worker_copy["workerPool"] = pool["id"]
+            mock_workers_flattened.append(worker_copy)
+    mock_get_workers.return_value = mock_workers_flattened
+
+    # get_runs returns flattened runs with stack field added
+    mock_runs_flattened = []
+    for stack in RUNS_DATA["data"]["stacks"]:
+        for run in stack.get("runs", []):
+            run_copy = run.copy()
+            run_copy["stack"] = stack["id"]
+            mock_runs_flattened.append(run_copy)
+    mock_get_runs.return_value = mock_runs_flattened
+
+    # get_entities returns flattened entities from all stacks
+    mock_entities_flattened = []
+    for stack in ENTITIES_DATA["data"]["stacks"]:
+        mock_entities_flattened.extend(stack.get("entities", []))
+    mock_get_entities.return_value = mock_entities_flattened
+
+    # Create mock Spacelift session using MagicMock
+    spacelift_session = MagicMock()
+
+    # Create mock EC2 instances (simulating AWS EC2 sync running before Spacelift sync)
+    for instance in EC2_INSTANCES_DATA:
+        neo4j_session.run(
+            """
+            MERGE (i:EC2Instance{instanceid: $instance_id})
+            SET i.id = $instance_id,
+                i.region = $region,
+                i.instancetype = $instance_type,
+                i.state = $state
+            """,
+            instance_id=instance["InstanceId"],
+            region=instance["Region"],
+            instance_type=instance["InstanceType"],
+            state=instance["State"],
+        )
```

Comment:
Ah like this is where we could just use this and run the actual EC2 sync to make the tests more robust: https://github.com/cartography-cncf/cartography/blob/master/tests/integration/cartography/intel/aws/ec2/test_ec2_instances.py#L22-L38

In case someone changes the EC2 sync in a way that breaks this in the future